README.md

```diff
@@ -506,6 +512,119 @@ $ mbedgt -V --target=frdm-k64f-gcc
+
+Changes:
+* New command line switch `--test-spec` is introduced. It is used to pass test specification file name to Greentea.
+* Existing command line switch `-t` together with `--test-spec` switch can be used to select build(s) by name which should be used for test runs. When no test specification is defined switch `-t` / `--target` behaves as usual: you can select yotta targets inside yotta module.
+
+## Test specification formatted
+
+More detailed test specification format will be introduced in near future. In current form test specification is a dictionary with key-value pairs under "builds" entry where key is a build name and value is a dictionary with additional properties describing build itself. Build properties include platform name, toolchain used to compile interface chipa baudrate and list of test binaries.
+
+## Example of test specification file
+
+In below example there are two build defined:
+* Build `K64F-ARM` for Freescale `K64F` platform compiled with `ARMCC` compiler and
+* build `K64F-GCC` for Freescale `K64F` platform compiled with `GCC ARM` compiler.
+
+```json
+{
+    "builds": {
+        "K64F-ARM": {
+            "platform": "K64F",
+            "toolchain": "ARM",
+            "base_path": "./.build/K64F/ARM",
+            "baud_rate": 115200,
+            "tests": {
+                "mbed-drivers-test-generic_tests":{
+                    "binaries":[
+                        {
+                            "binary_type": "bootable",
+                            "path": "./.build/K64F/ARM/mbed-drivers-test-generic_tests.bin"
+                        }
+                    ]
+                },
+                "mbed-drivers-test-c_strings":{
+                    "binaries":[
+                        {
+                            "binary_type": "bootable",
+                            "path": "./.build/K64F/ARM/mbed-drivers-test-c_strings.bin"
+                        }
+                    ]
+                }
+            }
+        },
+        "K64F-GCC": {
+            "platform": "K64F",
+            "toolchain": "GCC_ARM",
+            "base_path": "./.build/K64F/GCC_ARM",
+            "baud_rate": 115200,
+            "tests": {
+                "mbed-drivers-test-generic_tests":{
+                    "binaries":[
+                        {
+                            "binary_type": "bootable",
+                            "path": "./.build/K64F/GCC_ARM/mbed-drivers-test-generic_tests.bin"
+                        }
+                    ]
+                }
+
+            }
+        }
+
+    }
+}
+```
+
+In below examples we will use above test specification file.
+
+### Command line usage
+
+When building your mbed projects with <build system> capable of returning test specification in our format you can directly call Greentea to execute tests or list available tests (`-l` / `--list` switch).
```

Comment:
`<build system>` doesn't show up in rendered Markdown, should this be something else?